index.js

```diff
@@ -12,25 +13,37 @@ var isPromise = require('is-promise');
  */
 
 module.exports = function (func, cb) {
-  var async = false;
-  cb = once(cb);
+  return function () {
+    var async = false;
+    var promise = null;
+    cb = cb && once(cb);
+    if (typeof Promise !== 'undefined') {
+      promise = new Promise(function (resolve, reject) {
+        cb = promiseResolver(resolve, reject, cb);
+      });
+    } else if (!cb) {
+      throw new Error('no callback or global Promise implementation');
```

Comment:
Yep, makes sense. I can update shortly.

Do you think adding directions about setting global.promise seems overly verbose?